src/transformers/generation_utils.py

```diff
@@ -638,9 +641,18 @@ def _get_logits_processor(
             processors.append(InfNanRemoveLogitsProcessor())
         return processors
 
-    def _get_stopping_criteria(self, max_length: Optional[int], max_time: Optional[float]) -> StoppingCriteriaList:
-        stopping_criteria = StoppingCriteriaList()
-        if max_length is not None:
+    def _get_stopping_criteria(
+        self,
+        max_length: Optional[int],
+        max_time: Optional[float],
+        max_new_tokens: Optional[int],
+        start_length: int,
```

Comment:
start_length is not needed no?

Reply:
What is `max_new_tokens` needed for? I think this PR was from quite some time ago where we had this

Reply:
Yes, I think they come from my attempt to merge the main branch into this one. I'll fix this.